cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -0,0 +1,45 @@
+import * as express from 'express';
+import {UserSec} from '../user/UserSec';
+
+export interface Auth {
+  protect(role?: string): void;
+}
+
+export class PassportAuth implements Auth {
+  protected userSec: UserSec;
+  protected loginRoute: string;
+
+  constructor(protected readonly UserSecService: UserSec, loginRoute?: string) {
+    this.userSec = UserSecService;
+    this.loginRoute = loginRoute || '/login';
+  }
+
+  /**
+   * Function which checks if the user requesting access to the resource is authenticated and authorized to
+   * access the resource. Redirects to the login page if user is not authenticated or returns an 'Unauthorized'
+   * response if user does not have the required role.
+   *
+   * @param role {string} [Optional] Role which the user needs in order to access this resource
+   */
+  public protect(role?: string) {
+    return (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      if (req.isAuthenticated()) {
+        if (!role) {
+          return next();
+        } else {
+          this.userSec.hasResourceRole(role).then((hasRole: boolean) => {
+            return hasRole ? next() : res.status(401).send('Unauthorized');
```

Comment:
[Antipattern] Avoid sending strings like `'Unauthorized'` etc. This may break json parsers. 
Returning 401 it's enough. Body can be empty.